# Liang et al. 2025 unc-43 analysis

Analysis code for "The Challenge of Choosing a Negative Control: A Behavioural and Morphological Characterization of Various unc-43 LOF Alleles in the Model Organism C. elegans"

# Table of Contents
* [About the Project](#About-the-Project)
* [Prerequisites](#Prerequisites)
* [Getting Started](#Getting-Started)
* [Installation](#Installation)
* [Data](#Data)
* [Contact](#Contact)
* [Project Authors](#Project-Authors)
* [Acknowledgements](#Acknowledgements)

# About the Project
## Abstract
Loss-of-Function (LOF) analyses have been pivotal in the discovery of genetic function in biological processes. However, phenotypic variation can occur between different supposedly LOF alleles of the same gene. We investigated the body of research surrounding the nematode ortholog of the Calcium/Calmodulin-dependent protein Kinase II (CAMKII), unc-43, because of its relevance in a large range of biological functions. Our analysis shows that published findings on unc-43 function were obtained from studies using numerous loss-of-function alleles, leading to potential challenges in aligning research findings. We investigated the similarity of these putative loss-of-function (LOF) mutations by using the Multi-Worm Tracker (Swierczek et al., 2011) to phenotype nine LOF alleles across 27 phenotypes, spanning morphology, baseline locomotion and habituation learning and memory. Interestingly, our study reveals significant differences in phenotypes for these LOF alleles. From our data, we have identified the three putative LOF alleles with the most severe and similar phenotypes propose for two of them (n498n1186 and js125) to be regarded as reference alleles to streamline future research on unc-43 function. This data will help future researchers optimise their research of unc-43 function and determine how to choose a strain/strains to use as a negative control. 
# Getting Started

Make sure you have Python installed with the requirements listed below. Raw or preprocessed data can be obtained from the OSF repository for this project.

# Prerequisites

Coming soon....

# Installation

The Python packages required for this project are listed at the beginning of the Jupyter notebook in the first cell. 

Please report any missing dependencies/requirements by opening an issue on this repository.

# Data

Raw or preprocessed data can be obtained from the OSF repository for this project.

# Contact

For questions or comments specific to the implementation provided in this repository, please contact:

Joseph Liang (joseph.liang@psych.ubc.ca)

Additional questions about the project, such as further information and requests for resources and reagents should be directed to and will be fulfilled by the Lead Contact, Catharine H. Rankin (crankin@psych.ubc.ca).

# Project Authors

* Joseph J.H. Liang
* Anushka Sood
* Gayatree Khanna
* Catharine H. Rankin


# Acknowledgements

This work was supported by a Canadian Institutes of Health Research Discovery Grant #PJT 165974 to CHR, and a Natueral Sciences and Engineering Research Council of Canada (NSERC) Postgraduate Scholarship to JL. Strains were provided by the Caenorhabditis Genetics Center (NIH Office of the Director P40623 OD010440).
